modules/carto/src/api/layer-map.ts

```diff
@@ -341,15 +342,19 @@ export function getColorValueAccessor({name}, colorAggregation, data: any) {
 export function getColorAccessor(
   {name},
   scaleType: SCALE_TYPE,
-  {aggregation, range: {colors, colorMap}},
+  {aggregation, range: {colors, colorMap, type, column}},
   opacity: number | undefined,
   data: any
 ) {
-  const scale = SCALE_FUNCS[scaleType as any]();
+  let scale = SCALE_FUNCS[scaleType as any]();
   let domain: (string | number)[] = [];
   let scaleColor: string[] = [];
+  let accessorKeys = getAccessorKeys(name, aggregation);
 
-  if (Array.isArray(colorMap)) {
+  if (type === HEX_COLOR_COLUMN) {
```

Comment:
Rather than having the special case here, could you try adding [scaleIdentity](https://github.com/d3/d3-scale#scaleIdentity) to `SCALE_FUNCS` and then using that?

Reply:
Done here: [c53c724](https://github.com/visgl/deck.gl/pull/8177/commits/c53c724c4d87c479746af854840c63874061cc12)